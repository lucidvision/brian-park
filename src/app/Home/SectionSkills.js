import React from 'react'
import styled from 'styled-components'
import { HeadingSecondary, Skill } from 'components'
import { media, utilities } from 'styles'
import _ from 'lodash'

const StyledSection = styled.section`
  margin-top: -10rem;

  ${media.phone`
    margin-top: -4rem;
  `}
`

const StyledDiv = styled.div`
  display: flex;
  justify-content: space-between;

  ${media.phone`
    flex-direction: column;
    align-items: center;

    > * {
      &:not(:last-child) {
        margin-bottom: ${utilities.marginMedium};
      }
    }
  `}
`

const StyledHeadingSecondary = styled(HeadingSecondary)`
  text-align: center;
  margin-bottom: ${utilities.marginLarge};

  ${media.phone`
    margin-bottom: ${utilities.marginMedium};
  `}
`

const StyledSkill = styled(Skill)`
  width: 30%;

  ${media.phone`
    width: 90%;
  `}
`

const skills = [
  {
    title: 'Strategy',
    content: 'A big goal requires efficient planning. I use the 80/20 rule and iterative development to deliver exceptional results.'
  },
  {
    title: 'Product',
    content: 'Customer obsession inspires elegant design. I use feedback to craft interfaces that are easy to use.'
  },
  {
    title: 'Technology',
    content: 'Optimization drives continuous improvement. I use cutting-edge frameworks and tools to write clean, scalable code.'
  }
]

export default function SectionSkills (props) {
  return (
    <StyledSection>
      <StyledHeadingSecondary text='What I do' />
      <StyledDiv>
        {_.map(skills, (skill, index) => <StyledSkill key={index} {...skill} />)}
      </StyledDiv>
    </StyledSection>
  )
}
