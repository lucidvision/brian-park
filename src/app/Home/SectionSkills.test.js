import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import SectionSkills from './SectionSkills'

Enzyme.configure({adapter: new Adapter()})

const skills = [
  {
    title: 'Strategy',
    content: 'A big goal requires efficient planning. I use the 80/20 rule and iterative development to deliver exceptional results.'
  },
  {
    title: 'Product',
    content: 'Customer obsession inspires elegant design. I use feedback to craft interfaces that are easy to use.'
  },
  {
    title: 'Technology',
    content: 'Optimization drives continuous improvement. I use cutting-edge tools and frameworks to write clean, scalable code.'
  }
]

describe('SectionSkills', () => {
  const sectionSkills = shallow(<SectionSkills />)

  it('renders correctly', () => {
    expect(sectionSkills).toMatchSnapshot()
  })

  it('renders a secondary heading', () => {
    expect(sectionSkills.find('Styled(HeadingSecondary)').props().text).toBe('What I do')
  })

  describe('Skills', () => {
    it('renders 3 skills', () => {
      expect(sectionSkills.find('Styled(Skill)')).toHaveLength(3)
    })

    it('renders strategy skill', () => {
      expect(sectionSkills.find('Styled(Skill)').at(0).props()).toMatchObject(skills[0])
    })

    it('renders product skill', () => {
      expect(sectionSkills.find('Styled(Skill)').at(1).props()).toMatchObject(skills[1])
    })

    it('renders technology skill', () => {
      expect(sectionSkills.find('Styled(Skill)').at(2).props()).toMatchObject(skills[2])
    })
  })
})
