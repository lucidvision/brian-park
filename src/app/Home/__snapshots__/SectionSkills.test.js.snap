// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionSkills renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SectionSkills />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Styled(HeadingSecondary)
          text="What I do"
/>,
        <styled.div
          className="wow fadeInUp"
>
          <Styled(Skill)
                    content="A big goal requires efficient planning. I use the 80/20 rule and iterative development to deliver exceptional results."
                    title="Strategy"
          />
          <Styled(Skill)
                    content="Customer obsession inspires elegant design. I use feedback to craft interfaces that are easy to use."
                    title="Product"
          />
          <Styled(Skill)
                    content="Optimization drives continuous improvement. I use cutting-edge tools and frameworks to write clean, scalable code."
                    title="Technology"
          />
</styled.div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "text": "What I do",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Styled(Skill)
              content="A big goal requires efficient planning. I use the 80/20 rule and iterative development to deliver exceptional results."
              title="Strategy"
/>,
            <Styled(Skill)
              content="Customer obsession inspires elegant design. I use feedback to craft interfaces that are easy to use."
              title="Product"
/>,
            <Styled(Skill)
              content="Optimization drives continuous improvement. I use cutting-edge tools and frameworks to write clean, scalable code."
              title="Technology"
/>,
          ],
          "className": "wow fadeInUp",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "class",
            "props": Object {
              "content": "A big goal requires efficient planning. I use the 80/20 rule and iterative development to deliver exceptional results.",
              "title": "Strategy",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "content": "Customer obsession inspires elegant design. I use feedback to craft interfaces that are easy to use.",
              "title": "Product",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "class",
            "props": Object {
              "content": "Optimization drives continuous improvement. I use cutting-edge tools and frameworks to write clean, scalable code.",
              "title": "Technology",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(HeadingSecondary)
            text="What I do"
/>,
          <styled.div
            className="wow fadeInUp"
>
            <Styled(Skill)
                        content="A big goal requires efficient planning. I use the 80/20 rule and iterative development to deliver exceptional results."
                        title="Strategy"
            />
            <Styled(Skill)
                        content="Customer obsession inspires elegant design. I use feedback to craft interfaces that are easy to use."
                        title="Product"
            />
            <Styled(Skill)
                        content="Optimization drives continuous improvement. I use cutting-edge tools and frameworks to write clean, scalable code."
                        title="Technology"
            />
</styled.div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "text": "What I do",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Styled(Skill)
                content="A big goal requires efficient planning. I use the 80/20 rule and iterative development to deliver exceptional results."
                title="Strategy"
/>,
              <Styled(Skill)
                content="Customer obsession inspires elegant design. I use feedback to craft interfaces that are easy to use."
                title="Product"
/>,
              <Styled(Skill)
                content="Optimization drives continuous improvement. I use cutting-edge tools and frameworks to write clean, scalable code."
                title="Technology"
/>,
            ],
            "className": "wow fadeInUp",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "class",
              "props": Object {
                "content": "A big goal requires efficient planning. I use the 80/20 rule and iterative development to deliver exceptional results.",
                "title": "Strategy",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "class",
              "props": Object {
                "content": "Customer obsession inspires elegant design. I use feedback to craft interfaces that are easy to use.",
                "title": "Product",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "class",
              "props": Object {
                "content": "Optimization drives continuous improvement. I use cutting-edge tools and frameworks to write clean, scalable code.",
                "title": "Technology",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
